fgpyo/sam/__init__.py

```diff
@@ -547,6 +547,45 @@ def length_on_target(self) -> int:
         """Returns the length of the alignment on the target sequence."""
         return sum([elem.length_on_target for elem in self.elements])
 
+    def query_alignment_offsets(self) -> Tuple[int, int]:
+        """Gets the 0-based, end-exclusive positions of the first and last aligned base in the
+        query.
```

Comment:
I think this will fit on one line if you don't start it on the same line as the triple quote.

You could also omit the part about the positions being zero-based exclusive, and defer that detail to the body or the `Returns` section.

```suggestion
        """
        Gets the 0-based, end-exclusive positions of the first and last aligned base in the query.
```